golem-component-service-base/src/repo/component.rs

```diff
@@ -0,0 +1,648 @@
+                    c.name AS name,
+                    c.component_id AS component_id,
+                    cv.version AS version,
+                    cv.size AS size,
+                    cv.user_component AS user_component,
+                    cv.protected_component AS protected_component,
+                    cv.protector_version AS protector_version,
+                    cv.metadata AS metadata
+                FROM components c
+                    JOIN component_versions cv ON c.component_id = cv.component_id
+                WHERE c.namespace = $1 AND c.name = $2
+                "#,
+        )
+        .bind(namespace)
+        .bind(name)
+        .fetch_all(self.db_pool.deref())
+        .await
+        .map_err(|e| e.into())
+    }
+
+    async fn get_id_by_name(&self, namespace: &str, name: &str) -> Result<Option<Uuid>, RepoError> {
+        let result =
+            sqlx::query("SELECT component_id FROM components WHERE namespace = $1 AND name = $2")
+                .bind(namespace)
+                .bind(name)
+                .fetch_optional(self.db_pool.deref())
+                .await?;
+
+        Ok(result.map(|x| x.get("component_id")))
+    }
+
+    async fn get_namespace(&self, component_id: &Uuid) -> Result<Option<String>, RepoError> {
+        let result = sqlx::query("SELECT namespace FROM components WHERE component_id = $1")
+            .bind(component_id)
+            .fetch_optional(self.db_pool.deref())
+            .await?;
+
+        Ok(result.map(|x| x.get("namespace")))
+    }
+
+    async fn delete(&self, namespace: &str, component_id: &Uuid) -> Result<(), RepoError> {
+        let mut transaction = self.db_pool.begin().await?;
+        sqlx::query(
+            r#"
+                DELETE FROM component_versions
+                WHERE component_id IN (SELECT component_id FROM components WHERE namespace = $1 AND component_id = $2)
+            "#
+        )
+        .bind(namespace)
+        .bind(component_id)
+        .execute(&mut *transaction)
+        .await?;
+
+        sqlx::query("DELETE FROM components WHERE namespace = $1 AND component_id = $2")
+            .bind(namespace)
+            .bind(component_id)
+            .execute(&mut *transaction)
+            .await?;
+
+        transaction.commit().await?;
+        Ok(())
+    }
+}
+
+pub mod record_metadata_serde {
+    use bincode::{Decode, Encode};
+    use bytes::Bytes;
+    use golem_common::serialization::serialize_with_version;
+    pub const SERIALIZATION_VERSION_V1: u8 = 1u8;
+
+    pub fn serialize<T: Encode>(routes: &T) -> Result<Bytes, String> {
+        serialize_with_version(routes, SERIALIZATION_VERSION_V1)
+    }
+
+    pub fn deserialize<T: Decode>(bytes: &[u8]) -> Result<T, String> {
+        let (version, data) = bytes.split_at(1);
+
+        match version[0] {
+            SERIALIZATION_VERSION_V1 => {
+                let (routes, _) = bincode::decode_from_slice(data, bincode::config::standard())
```

Comment:
This is not routes

Reply:
this was just typo, serialization changed to proto